standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/metrics/Metrics.java

```diff
@@ -160,6 +159,47 @@ public Integer getValue() {
     }
   }
 
+  /**
+   * Get a Map that represents a multi-field metric,
+   * or create a new one if it does not already exist.
+   * @param name Name of map metric.  This should come from MetricConstants
+   * @return MapMetric .
+   */
+  public static MapMetrics getOrCreateMapMetrics(String name) {
+    if (self == null) {
+      return dummyMapMetrics;
+    }
+
+    Map<String, Metric> metrics = self.registry.getMetrics();
+    Metric map = metrics.get(name);
+    if (map instanceof MapMetrics) {
+      return (MapMetrics) map;
+    }
+
+    // Looks like it doesn't exist.  Lock so that two threads don't create it at once.
+    synchronized (Metrics.class) {
+      // Recheck to make sure someone didn't create it while we waited.
```

Comment:
The recheck is missing. To follow the pattern of the other metrics, this block should look like:

```
metrics = self.registry.getMetrics();
      map = metrics.get(name);
      if (map != null && map instanceof MapMetrics) {
        return (MapMetrics) map;
      }
// then register the metric and return it
```

Reply:
The `instanceof` is nullsafe

Reply:
```java
    Object val = null;
    if (val instanceof String) {
      LOG.info("Not null safe");
    } else {
      LOG.info("Null safe");
    }
```
output
```log
[INFO ] OMITTED - Null safe
```

Reply:
ohh sorry I've missed that

Reply:
Fixed and thanks